# HG changeset patch
# Parent 65fb8b9ea0b702518cf63bae97fb703222ffa7b1
# User Chris Coulson <chris.coulson@canonical.com>
Fix make-makefile test failure when the source path contains perl regexp metacharacters

diff --git a/build/autoconf/test/makemakefile.tpm b/build/autoconf/test/makemakefile.tpm
--- a/build/autoconf/test/makemakefile.tpm
+++ b/build/autoconf/test/makemakefile.tpm
@@ -115,17 +115,17 @@ sub check_getTopDir
     print "Running: check_getTopDir\n" if ($main::argv{debug});
 
     my $path = getTopDir();
 
     ## Unit test is special, cmd not invoked from the same directory
     ## as the makemakefile.pm module.
     ok($path ? 1 : 0, 1, "getTopDir failed");
     ok(-d $path ? 1 : 0, 1, "getTopDir: directory $path does not exist");
-    ok($FindBin::RealBin =~ m%$path/% ? 1 : 0, 1, 'Invalid topdir path');
+    ok($FindBin::RealBin =~ m%\Q$path/% ? 1 : 0, 1, 'Invalid topdir path');
     ok(-e "$path/client.mk" ? 1 : 0, 1, "client.mk not found in $path");
 } # check_getTopDir
 
 ###########################################################################
 ## Intent: Verify objdir lookup function
 ###########################################################################
 sub check_getObjDir
 {

